### PR TITLE
new `setnick` command with hierarchy check.

### PR DESCRIPTION
**Pull request by DTS-11**

### CODE REVIEW

**Review by realstealthninja — APPROVED** [review]
everything looks fine.
few notes:
 - it would have been nice if it was an embed
 - making it delete after a certain time would also be a good addition
other than this everything is good! 👍

**Comment by DTS-11** [comment]
> everything looks fine. few notes:
> 
> * it would have been nice if it was an embed
> * making it delete after a certain time would also be a good addition
>   other than this everything is good! 👍

does other commands delete on its own?

**Comment by realstealthninja** [comment]
why i mean by deleting is adding `delete_after=10`
```py
await ctx.send(f'Nickname for {member.name} was changed to {member.mention}', delete_after=10)
```

**Comment by DTS-11** [comment]
> why i mean by deleting is adding `delete_after=10`
> 
> ```python
> await ctx.send(f'Nickname for {member.name} was changed to {member.mention}', delete_after=10)
> ```

for now, no need of embed.. anyway its gonna delete its msg

**Comment by realstealthninja** [comment]
very good!

**Review by realstealthninja — APPROVED** [review]

File: src/cogs/mod.py [hunk 1]
```diff
@@ -47,6 +47,19 @@ def parse(self, content: str) -> int:
...
+    @commands.command()
+    @commands.guild_only()
+    @commands.has_permissions(manage_nicknames=True)
+    @commands.bot_has_permissions(manage_nicknames=True)
+    async def setnick(self, ctx, member: disnake.Member, *, nick):
+        """
+        Set a custom nick-name.
+        """
+        if not ctx.author.top_role.position > member.top_role.position:
+            return await ctx.send("You cant change someone's nickname that is higher or the same role heirarchy.")
```
Comment (Yat-o) [inline src/cogs/mod.py:60]: This doesn’t seem like a right error message to put out. The logical check you added only passes if they are only greater than the commands author. Not if they are greater or equal to.

File: src/cogs/mod.py [hunk 2]
```diff
@@ -47,6 +47,19 @@ def parse(self, content: str) -> int:
...
+        await member.edit(nick=nick)
```
Comment (Yat-o) [inline src/cogs/mod.py:61]: Inconsistent use of quotes. Black your code.

File: src/cogs/mod.py [hunk 3]
```diff
@@ -47,6 +47,19 @@ def parse(self, content: str) -> int:
         elif duration == "h":
             seconds = int(seconds) * 3600
         return seconds
+    
+    @commands.command()
+    @commands.guild_only()
+    @commands.has_permissions(manage_nicknames=True)
+    @commands.bot_has_permissions(manage_nicknames=True)
+    async def setnick(self, ctx, member: disnake.Member, *, nick):
```
Comment (Yat-o) [inline src/cogs/mod.py:55]: Type hint `ctx` to `commands.Context`

**Review by Yat-o — CHANGES_REQUESTED** [review]
It was also be imperative to check how the length of how land characters `nick` is. If you try to change someone’s name to something over 36(I think) characters. It will throw an HTTPException.

File: src/cogs/mod.py (hunk 2 above)
Comment (DTS-11) [reply src/cogs/mod.py:61]: wdym?

Comment (Yat-o) [reply src/cogs/mod.py:61]: You're using single quotes where you should be using double quotes for consistency.

Comment (DTS-11) [reply src/cogs/mod.py:61]: > You're using single quotes where you should be using double quotes for consistency.

which line?

Comment (Yat-o) [reply src/cogs/mod.py:61]: 62

**Comment by DTS-11** [comment]
> It was also be imperative to check how the length of how land characters `nick` is. If you try to change someone’s name to something over 36(I think) characters. It will throw an HTTPException.

done..

File: src/cogs/mod.py (hunk 1 above)
Comment (Yat-o) [reply src/cogs/mod.py:60]: This still hasn’t been resolved for conciseness

**Review by Yat-o — CHANGES_REQUESTED** [review]

File: src/cogs/mod.py (hunk 1 above)
Comment (Kirk-KD) [reply src/cogs/mod.py:60]: It is correct, it says if the sender's top role is NOT greater than the other user, which means smaller or equal to.

Comment (Yat-o) [reply src/cogs/mod.py:60]: Look at the logical check again. It only checks for greater than.

Comment (Yat-o) [reply src/cogs/mod.py:60]: Oh yeah. I just now saw the `not`

**Review by Yat-o — APPROVED** [review]

**Pull request merged**